### PR TITLE
Add mod3 and mod4 as modifier keys

### LINKED ISSUE #317: Request Support for Mod3 (and more)
Support for further modkeys would be useful, and put Amethyst at feature parity for keybindings with a lot of other Linux Tiling WMs. I switch between my home machine (AwesomeWM) and my work machine (Amethyst) and would love to be able to mirror my keybinding setup from Awesome on my work machine.

[Trello Card](https://trello.com/c/MOcMvs8I/146-amethyst-request-support-for-mod3-and-more)

### PR DESCRIPTION
**Pull request by rjuer**
Closes #317

### CODE REVIEW

**Comment by ianyh** [comment]
Cool. I'd rather not define them by default, though.

**Comment by rjuer** [comment]
That's a very reasonable suggestion - I added the fix.

File: Amethyst/Preferences/UserConfiguration.swift [hunk 1]
```diff
@@ -363,9 +367,13 @@ class UserConfiguration: NSObject {
 
         let mod1Strings: [String] = configurationValueForKey(.mod1)!
         let mod2Strings: [String] = configurationValueForKey(.mod2)!
+        let mod3Strings: [String] = configurationValueForKey(.mod3)!
```
Comment (ianyh) [inline Amethyst/Preferences/UserConfiguration.swift]: Ah, I see why the `[]`. I think we can remove the lines from `default.amethyst` entirely and remove the force unwrap here? As it is now, technically something like `mod3+x` would just get triggered by `x`.

**Review by ianyh — CHANGES_REQUESTED** [review]

File: Amethyst/Preferences/UserConfiguration.swift [hunk 2]
```diff
@@ -363,9 +367,18 @@ class UserConfiguration: NSObject {
 
         let mod1Strings: [String] = configurationValueForKey(.mod1)!
         let mod2Strings: [String] = configurationValueForKey(.mod2)!
+        let mod3Strings: [String]? = configurationValueForKey(.mod3)
+        let mod4Strings: [String]? = configurationValueForKey(.mod4)
 
         modifier1 = modifierFlagsForStrings(mod1Strings)
         modifier2 = modifierFlagsForStrings(mod2Strings)
+
+        if mod3Strings != nil {
```
Comment (ianyh) [inline Amethyst/Preferences/UserConfiguration.swift]: The swift idiom here would be unwrapping.
```swift
if let modStrings = mod3Strings {
    // modStrings is a regular value in scope
}
```

**Review by ianyh — CHANGES_REQUESTED** [review]

**Review by ianyh — APPROVED** [review]

**Comment by ianyh** [comment]
Thanks for the contribution!

**Pull request merged**